Add size to constant - type dispatche through BaseListVariable.cls_for

voznesenskym opened the pull request:
Summary: Fixes a small bug

Test Plan: Ran failing tests

Reviewed By: PoojaAg18

Differential Revision: D49662177




cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

ezyang commented on torch/_dynamo/variables/constant.py:
```diff
@@ -104,7 +104,8 @@ def getitem_const(self, arg: VariableTracker):
             return True
-        if type(obj) in (list, tuple, set, frozenset):
+        # The structure within is_literal get routed to variables.BaseListVariable
+        if type(obj) in (list, tuple, set, frozenset, torch.Size):
```
note that unlike the other container types, entries in torch.Size can *only* be ints (other entries are rejected). This has caused not considerable grief when dealing with SymInts/pytree'ing. I don't know if it is relevant here; it is hard to tell what the downstream effects are.

@jansel, ngl, I still have no idea what a "literal" is, in your definition haha.

voznesenskym replied:
This should be safe, protected during constant construction

voznesenskym commented:
This unblocks a failing internal test. I am working on a repro in parallel.

Commit:
Add size to constant - type dispatche through BaseListVariable.cls_for (#110126)

Summary:

Fixes a small bug

Test Plan: Ran failing tests

Reviewed By: PoojaAg18, yanboliang

Differential Revision: D49662177
